packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenModule.kt

```diff
@@ -0,0 +1,52 @@
+package expo.modules.splashscreen
+
+import android.content.Context
+
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ActivityProvider
+import org.unimodules.core.interfaces.ExpoMethod
+
+private const val NAME = "ExpoSplashScreen"
+private const val ERROR_TAG = "ERR_SPLASH"
+
+class SplashScreenModule(context: Context) : ExportedModule(context) {
+  private lateinit var activityProvider: ActivityProvider
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    activityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
+  }
+
+  @ExpoMethod
+  fun preventAutoHideAsync(promise: Promise) {
+    val activity = activityProvider.currentActivity
+    if (activity == null) {
+      promise.reject(ERROR_TAG, "No valid activity.")
+      return
+    }
+    SplashScreen.preventAutoHide(
```

Comment:
If `@JvmStatic` magically ensures that `SplashScreen.preventAutoHide` would be called on a single instance of `SplashScreen`, couldn't we grab the `SplashScreen` instance from the `moduleRegistry`? I wish we could…

Reply:
It's not about `@JvmStatic` annotation. It's about `SplashScreen` being an `object`. And yes it ensures you that it's always the very same instance and the only one.
I can go your way (we actually have this already this way in iOS, but please provide me either `a)` a way to do so or `b)` right to do it so, but after this initial `expo-splash-screen` lands 😉